splunk-ingestable-s3-bucket/main.tf

```diff
@@ -0,0 +1,210 @@
+            "StringEquals":
+                {
+                "aws:PrincipalOrgID":["${var.aws_organization_id}"]
+                }
+         },
+         "Resource": [
+            "arn:aws:s3:::${var.log_bucket_name}/*"
+         ]
+      }
+   ]
+}
+POLICY
+}
+
+// S3 Bucket event notification that gets triggered on new items
+
+resource "aws_s3_bucket_notification" "object_created" {
+  bucket = "${aws_s3_bucket.this.id}"
+
+  topic {
+    topic_arn = "${aws_sns_topic.bucket_events.arn}"
+
+    events = [
+      "s3:ObjectCreated:*",
+    ]
+
+    filter_suffix = ""
+  }
+
+  depends_on = [
+    "aws_sns_topic.bucket_events",
+  ]
+}
+
+resource "aws_sns_topic" "bucket_events" {
+  name = "s3-notification-topic-${var.log_bucket_name}"
+}
+
+resource "aws_sns_topic_policy" "default" {
+  arn    = "${aws_sns_topic.bucket_events.arn}"
+  policy = "${data.aws_iam_policy_document.sns_policy.json}"
+}
+
+data "aws_iam_policy_document" "sns_policy" {
+  statement {
+    actions = [
+      "SNS:Publish",
+    ]
+
+    effect = "Allow"
+
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+
+      values = [
+        "${aws_s3_bucket.this.arn}",
+      ]
+    }
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    resources = [
+      "${aws_sns_topic.bucket_events.arn}",
+    ]
+
+    sid = "allowpublish"
+  }
+
+  statement {
+    actions = [
+      "SNS:Subscribe",
+      "SNS:Receive",
+    ]
+
+    effect = "Allow"
+
+    principals {
+      type        = "AWS"
+      identifiers = ["*"]
+    }
+
+    condition {
+      test     = "ArnEquals"
+      values   = ["${aws_sqs_queue.new_objects.arn}"]
+      variable = "aws:SourceArn"
+    }
+
+    resources = [
+      "${aws_sns_topic.bucket_events.arn}",
+    ]
+
+    sid = "sid_allow_subscribe"
+  }
+}
+
+// Queue that Splunk will subscribe to
+resource "aws_sqs_queue" "new_objects" {
+  name                       = "new-objects-for-${var.log_bucket_name}"
+  visibility_timeout_seconds = "${var.sqs_visibility_timeout_seconds}"
+  message_retention_seconds  = "${var.sqs_message_retention_seconds}"
+  receive_wait_time_seconds  = "${var.sqs_receive_wait_time_seconds}"
+  redrive_policy             = "{\"deadLetterTargetArn\":\"${aws_sqs_queue.dlq.arn}\",\"maxReceiveCount\":4}"
+}
+
+data "aws_iam_policy_document" "sqs_queue_policy" {
+  statement {
+    effect = "Allow"
+
+    principals {
+      identifiers = [
+        "*",
+      ]
+
+      type = "AWS"
+    }
+
+    actions = [
+      "SQS:SendMessage",
+    ]
+
+    resources = [
+      "${aws_sqs_queue.new_objects.arn}",
+    ]
+
+    condition {
+      test = "ArnEquals"
+
+      values = [
+        "${aws_sns_topic.bucket_events.arn}",
+      ]
+
+      variable = "aws:SourceArn"
+    }
+  }
+}
+
+// Dead Letter queue, use same paras as main queue
+resource "aws_sqs_queue" "dlq" {
+  name                      = "application-logs-dlq-${var.log_bucket_name}"
```

Comment:
Maybe it should not be hardcoded to `application-logs`, but `dlq-`

Reply:
Absoutely